lib/resource-mapper.js

```diff
@@ -84,28 +84,29 @@ class ResourceMapper {
 
   // Maps the request for a given resource and representation format to a server file
   // Will look for an index file if a folder is given and searchIndex is true
-  async mapUrlToFile ({ url, contentType, createIfNotExists, searchIndex = true }) {
+  async mapUrlToFile ({ url, contentType, createIfNotExists, searchIndex }) {
     // Parse the URL and find the base file path
     const { pathname, hostname } = this._parseUrl(url)
     const filePath = this.resolveFilePath(hostname, decodeURIComponent(pathname))
     if (filePath.indexOf('/..') >= 0) {
       throw new Error('Disallowed /.. segment in URL')
     }
-    let isIndex = searchIndex && filePath.endsWith('/')
+    let isFolder = filePath.endsWith('/')
+    let isIndex = searchIndex && isFolder
```

Comment:
I think this variable can be dropped (as per my comment below).

Reply:
Not being a coder. I need help.

- L127 OK `searchIndex` and drop of `isIndex` in L95
- L94 do you want to replace `isFolder` by `filePath.endsWith('/') `  everywhere it's used or to use an other variable name or to pass a parameter (this test was already used)
(The code allready used the endsWith('/') I would like to avoid at this stage to add a new parameter and push a test on all `mapUrlToFile` calls)

- I suppose I have to modify the tests that errored N°5 and 6 in travis due to my modifications. It may relates to explicitely set searchIndex value to true.

How can I send my modifications : as a new PR or something else (I have never done it)
Sorry for my deeply need of help

Reply:
> Not being a coder.

This is good stuff though 👍

L94, okay, leave as is.

> How can I send my modifications

You just push another commit to the branch you have created. So just keep pushing to your own `master` branch, and they will end up here. Don't hesitate to ask if you're uncertain, happy to explain!

Reply:
**ressource-mapper tests**
I have a problem with that test : 

>     itMapsUrl(mapper, 'a URL ending with a slash to a folder when index is skipped',
      {
        url: 'http://localhost/space/',
        contentType: 'application/octet-stream',
        createIfNotExists: true,
        searchIndex: false
      },
      {
        path: `${rootPath}space/`,
        contentType: 'application/octet-stream'
      })

Is it a use case ? 
It creates a folder with contentType.
In all tests cases it is the only case where a folder is created. I missed this behaviour.
If yes I need to slightly modify my code and I did not found where it is used (PUT, PATCH, POST)

Can you confirm.

Reply:
Is tthere no code calling it with `createIfNotExists: true` and `searchIndex: false`? Because this code might be used to create a folder.